sympy/polys/ring_series.py

```diff
@@ -1245,8 +1266,13 @@ def rs_tan(p, x, prec):
                 c_expr = c.as_expr()
                 const = R(tan(c_expr))
             except ValueError:
-                    raise DomainError("The given series can't be expanded in "
-                                      "this domain.")
+                R = R.add_gens([tan(c_expr, )])
+                #gens = list(R.gens).remove(x.set_ring(R))
+                #R.drop_to_ground(gens)
```

Comment:
Either remove them or document that this is another option to consider and pros/cons.